### PR TITLE
Add MINTTL parameter to cache configuration.

### PR DESCRIPTION
**Pull request by aaronriekenberg**
### 1. Why is this pull request needed and what does it do?
Adds a MINTTL parameter to allow specifying a minimum ttl for success and denial responses that can override the ttl for the backend.  This can be useful for limiting queries to the backend.

### 2. Which issues (if any) are related?
https://github.com/coredns/coredns/issues/2054

### 3. Which documentation changes (if any) need to be made?
PR updates cache/plugin/README.md

### CODE REVIEW

**Review by grobie — COMMENTED** [review]
Why didn't you add the minttl option to negative caches as well?

**Comment by chrisohaver** [comment]
> Why didn't you add the minttl option to negative caches as well?

This would be a good option.  Specifically as a means to mitigate k8s n-dots/search-path NXDOMAIN spamming issue.

**Comment by aaronriekenberg** [comment]
Added minttl option to negative cache also.

File: plugin/cache/cache.go [hunk 1]
```diff
@@ -100,6 +104,17 @@ func hash(qname string, qtype uint16, do bool) uint32 {
 	return h.Sum32()
 }
 
+func computeTTL(msgTTL time.Duration, minTTL time.Duration, maxTTL time.Duration) (ttl time.Duration) {
```
Comment (miekg) [inline plugin/cache/cache.go]: you don't need to repeat time.Duration for all these parameters

File: plugin/cache/cache.go [hunk 2]
```diff
@@ -100,6 +104,17 @@ func hash(qname string, qtype uint16, do bool) uint32 {
...
+	ttl = msgTTL
+	if ttl < minTTL {
+		ttl = minTTL
+	}
+	if ttl > maxTTL {
+		ttl = maxTTL
+	}
+	return
```
Comment (miekg) [inline plugin/cache/cache.go]: no naked returns please, name the variable

File: plugin/cache/cache.go [hunk 3]
```diff
@@ -228,7 +241,9 @@ func (w *ResponseWriter) Write(buf []byte) (int, error) {
 
 const (
 	maxTTL  = dnsutil.MaximumDefaulTTL
+	minTTL  = 0
 	maxNTTL = dnsutil.MaximumDefaulTTL / 2
+	minNTTL = 0
```
Comment (miekg) [inline plugin/cache/cache.go]: in pkg/dnsutil/ttl.go we say the minimum ttl coredns allows is 5s, which may not be (currently) true, but certainly clashes with these values.

File: plugin/cache/cache.go (hunk 1 above)
Comment (aaronriekenberg) [reply plugin/cache/cache.go]: Fixed

File: plugin/cache/cache.go (hunk 2 above)
Comment (aaronriekenberg) [reply plugin/cache/cache.go]: Fixed

File: plugin/cache/cache.go (hunk 3 above)
Comment (aaronriekenberg) [reply plugin/cache/cache.go]: Set minTTL and minNTTL to dnsutil.MinimalDefaultTTL.

**Review by miekg — APPROVED** [review]

**Review by grobie — APPROVED** [review]

**Pull request merged**